command/src/state.rs

```diff
@@ -29,6 +27,47 @@ use crate::{
 /// To use throughout Sōzu
 pub type ClusterId = String;
 
+/// Used only when returning errors
+#[derive(Debug)]
+pub enum ObjectKind {
+    Cluster,
+    Listener,
+    Backend,
+    HttpFrontend,
+    HttpsFrontend,
+    TcpFrontend,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum StateError {
+    #[error("Request came in empty")]
+    EmptyRequest,
+    #[error("dispatching this request did not bring any change to the state")]
+    NoChange,
+    #[error("State can not handle this request")]
+    UndispatchableRequest,
+    #[error("Did not find object with address or id '{0}'")]
+    NotFound(String),
```

Comment:
```suggestion
    NotFound(ObjectKind, string),
```
Am I missing a thing ?

Reply:
I hope not, I'll take that into account